api/python/quilt3/packages.py

```diff
@@ -865,35 +863,35 @@ def build(self, name=None, registry=None, message=None):
         Returns:
             The top hash as a string.
         """
-        return self._build(name=name, registry=registry, message=message)
-
-
-    def _build(self, name=None, registry=None, message=None):
-        self._set_commit_message(message)
+        validate_package_name(name)
 
         if registry is None:
             registry = get_from_config('default_local_registry')
         else:
             registry = fix_url(registry)
 
-        registry = registry.rstrip('/')
-        validate_package_name(name)
+        registry_parsed = PhysicalKey.from_url(registry)
```

Comment:
Similarly this should be in `_build` so that `build` and `build` take in the same argument types.